UI-2391 Extend px-map-marker-group component to support numbered marker

mallik-svmx opened the pull request:
- Re-defined structure to support rendering numbered marker:
`"marker-icon": {
  "icon-base": "custom-icon",
  "icon-type": "px-nav:unconfirmed", //also component defaults to px-nav:unconfirmed
  "icon-strokeColor": "blue",
  "icon-fillColor": "white",
  "icon-content": "1", //"icon-content": "px-nav:home" for icon marker
  "icon-options": { //currently it doesn't work, will be taken after/or part of UI-2363
    "strokeColor": "blue",
    "fillColor": "white",
  }
}`
- Icon-options is only valid when an "icon-content": "px-nav:home" and not a number.
- Enhanced demo to illustrate the number markers with marker group.
- Updated release notes.
- Reference demo screenshots.
<img width="615" alt="screen shot 2018-08-17 at 12 09 23 pm" src="https://user-images.githubusercontent.com/34659014/44284542-ed5bd980-a216-11e8-91a6-18b5d402aa56.png">
<img width="616" alt="screen shot 2018-08-17 at 1 12 45 pm" src="https://user-images.githubusercontent.com/34659014/44287055-3cf2d300-a220-11e8-9eed-1680f8c4ed6a.png">

mallik-svmx commented:
Addressed. @shao

shao commented:
Looks better - it still looks a little bit off-center to the left?  If you expand the image and you can see this?

mallik-svmx commented:
Good catch :-) @shao Now it's fixed using text-align center vs. margin props.

+ Added custom icon on to clustered marker and updated Jira with new marker-icon Json. @sabrinalui-ge thank you! for calling that out.

mallik-svmx commented on HISTORY.md:
```diff
@@ -1,3 +1,8 @@
+v3.6.1
```
Done

mallik-svmx commented on px-map-behavior-icon.es6.js:
```diff
@@ -251,4 +251,66 @@
+
+    createIcon(settings={}) {
+      // type defaults to type='px-nav:unconfirmed'
+      let { type='px-nav:unconfirmed', styleScope, strokeColor, fillColor, content, options: { icon, strokeColor: iconStrokeColor, fillColor: iconFillColor } } = settings;
```
Fixed

hanzge commented:
Merging with 2 approvals.
